src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2377,6 +2377,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 {
                     Error(diagnostics, ErrorCode.ERR_RefAssignmentMustHaveIdentityConversion, expression.Syntax, targetType);
                 }
+                else
+                {
+                    return expression;
+                }
```

Comment:
The behavior is the same as with byref arguments.I do not think Ref returns should not be any different.
We validate that an identity conversion exists but do not create a conversion.

CreateConversion is used to generate nodes that convert values. It should not be called here, because we do not convert values.

---
In reply to: [81204219](https://github.com/dotnet/roslyn/pull/14176#discussion_r81204219) [](ancestors = 81204219)